Some small usability fixes

learn-more opened the pull request:
- Sort the table view on the Sequence column (when it is present)
- Convert the table dropdown to a DropDownList
- Autosize the columns in the property grid
- Autosize the columns in the table grid
- Add .gitignore entry for .vs dir

Split up in individual commits per feature, if you disagree with any of those please let me know and I can drop it from the PR.

learn-more commented on src/LessMsi.Gui/MainForm.cs:
```diff
@@ -184,13 +184,43 @@ public void SetTableViewGridDataSource(IEnumerable<object[]> values)
 			}
+			if (msiTableGrid.IsHandleCreated)
+			{
+				msiTableGrid.AutoResizeColumns();
```
Another option would be to set the property for this, but that slows down scrolling in the control.

activescott replied:
If the slowdown isn't significant I prefer the property to reduce the overall code complexity. If the slowdown is significant (use your judgement on "significant"), I'm fine with finding a place to make the explicit method call. Seems unnatural to have to add such an event handler just to get the columns to sort, but if that's the way it is 🤷

learn-more replied:
The property was the first thing I did, but the scrolling was just annoying enough for me to spend the time to do it this way.

activescott commented:
Thanks again @learn-more! I don't see anything wrong here, but I'll have to take a closer look at this one and play with it. Again, I should be able to either merge it or give you some feedback this week or over the upcoming weekend.

activescott commented on src/LessMsi.Gui/MainFormPresenter.cs:
```diff
@@ -433,6 +433,22 @@ private void UpdateMSiTableGrid(Database msidb, string tableName)
 					// NOTE: Deliberately not calling msidb.TableExists here as some System tables could not be read due to using it.
 					string query = string.Concat("SELECT * FROM `", tableName, "`");
 
+					bool hasSequence = false;
```
@learn-more One idea here to improve this would be to instead add the sorting logic into the view layer. This would prevent opening and close, and re-opening that database here. I think you could make it as simple as adding a call like this below just before the `Status();` call `this.View.TableViewSortBy("Sequence");` Then implement `TableViewSortBy` on `IMainFormView`/`MainForm` and if the specified column name doesn't exist just ignore it. The DataGrid can be sorted with [DataGridView.Sort Method](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.datagridview.sort?view=netframework-4.0#System_Windows_Forms_DataGridView_Sort_System_Windows_Forms_DataGridViewColumn_System_ComponentModel_ListSortDirection_). 

I haven't tried this so it's all theory, but I think that would be better than baking the SQL, re-opening of the DB, etc. behavior into this later. If you have concerns or think it's too much to ask, lets discuss!

learn-more replied:
@activescott I'll have a go at that, will let you know how it works out.
If it works out fine, I'll force-push to this PR so that there is a clean commit history.

In my other PR you mentioned 'conventional commit syntax in the commit message.'
Is that something that I should do in my commits?

activescott replied:
> In my other PR you mentioned 'conventional commit syntax in the commit message.'
> Is that something that I should do in my commits?

It is ideal, but not required since I can squash-merge a PR and clean up history that way so not a /huge/ deal but all of this will be merged as a single commit and a single feature (e.g. "feat: usability improvements", albeit with a description). I think even more valuable than worrying about the commit history is to submit smaller PRs - like one for each feature/bug. I really like collaborating with others on this type of thing though, so I'll work through it with you in any case. Thanks again!!

learn-more replied:
> > In my other PR you mentioned 'conventional commit syntax in the commit message.'
> > Is that something that I should do in my commits?
> 
> It is ideal, but not required since I can squash-merge a PR and clean up history that way so not a /huge/ deal but all of this will be merged as a single commit and a single feature (e.g. "feat: usability improvements", albeit with a description). I think even more valuable than worrying about the commit history is to submit smaller PRs - like one for each feature/bug. I really like collaborating with others on this type of thing though, so I'll work through it with you in any case. Thanks again!!

Due to the size of the individual issues I dismissed that idea, but if you prefer smaller single feature PR's I will try to keep that in mind 👍

activescott commented on src/LessMsi.Gui/MainFormPresenter.cs:
```diff
@@ -433,6 +433,22 @@ private void UpdateMSiTableGrid(Database msidb, string tableName)
+					bool hasSequence = false;
+					using (var view = new ViewWrapper(msidb.OpenExecuteView(query)))
+					{
+						foreach (ColumnInfo col in view.Columns)
```
Not an issue, more of a nit, but you can avoid the verbosity of a for foreach with [C#'s linq syntax](http://csharp.academy/linq-where/) in instances like this. For example, I think the equivalent of your foreach block here would be `hasSequence = view.Columns.Where((col) => col.Name == "Sequence").Count() > 0;`

activescott commented on src/LessMsi.Gui/MainForm.cs:
```diff
@@ -192,21 +192,35 @@ public void SetTableViewGridDataSource(IEnumerable<object[]> values)
 		}
 
-		private void MsiTableGrid_HandleCreated(object sender, EventArgs e)
+		private void Grid_HandleCreated(object sender, EventArgs e)
```
Rather than add such a specific event and casting here, could we add this to another place where we're already initializing things? Like maybe the end of `MainForm.LoadFile` or something. Isn't that a safe time to do this or am I missing something?

activescott replied:
Now I see that you're using the Grid_HandleCreated event for two different grids. If we need it for one, I'm fine having it for both. Maybe worth a comment in that handler that it could be either grid though.

activescott reviewed (changes requested):
Please look over my comments and let me know if you think I should let it ride or if you want to make changes. The only one I'd say I'd _like_ you to consider changing is moving the sorting logic into the view layer. The other ones were mostly a nit. Codacy also found a nit in that code that would be ideal if you can fix that too: https://app.codacy.com/gh/activescott/lessmsi/pullRequest?prid=6748020

I'll work with you though. I am grateful for your contribution and I want to support it! 🙏  So happy to lend a hand or discuss things if you have a different point of view than I do.

learn-more commented:
@activescott I have addressed your comments.

For the sorting: I did agree that this was needlessly verbose in the place where it was, so I moved the code to an extension function.
The sorting is now down in the way you suggested.

activescott commented:
Dope. Love the DataGridviewExtentionMethods approach. Well done!

activescott commented:
:tada: This PR is included in version 1.8.0 :tada:

The release is available on:
- `v1.8.0`
- [GitHub release](https://github.com/activescott/lessmsi/releases/tag/v1.8.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
